### PR TITLE
Update action bar to the new Material style

### LINKED ISSUE #164: Update action bar to material design

### PR DESCRIPTION
**Pull request by HenriqueRocha**
The main purpose of this PR is to close #164.
I'll implement #174 on top of this one if it gets merged.

Some refactoring and code formatting changes were done along the way while implementing this.

### CODE REVIEW

**Comment by HenriqueRocha** [comment]
Looks like the build is still throwing java.lang.OutOfMemoryError: Java heap space :(

**Comment by stefanhoth** [comment]
@HenriqueRocha  So, I found another setting that seems to help the build. Could you merge this change in and let the CI retry?

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,131 +1,193 @@
...
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
-    <uses-permission android:name="android.permission.USE_CREDENTIALS"/>
-    <uses-permission android:name="android.permission.GET_ACCOUNTS" />
-    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
-    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
-    <uses-permission android:name="android.permission.WAKE_LOCK" />
-    <uses-permission android:name="android.permission.NFC" />
-    <uses-permission android:name="android.permission.GET_TASKS" />
+  <supports-screens
+    android:normalScreens="true"
```
Comment (tasomaniac) [inline app/src/main/AndroidManifest.xml]: I am told that using this is wrong. Because it disables out some devices and possible devices in the future. Like xxlarge screens for example.

Comment (stefanhoth) [inline app/src/main/AndroidManifest.xml]: > This attribute is "true" by default.
[Doc](http://developer.android.com/guide/topics/manifest/supports-screens-element.html#normal)

:+1: for removing it as it is redundant.

Same goes for `smallScreens`

**Comment by tasomaniac** [comment]
I reviewed the code. I saw a really good quality refactoring :smile: with lots of useful stuff. 

since the appcompat integration and refactoring are all together, this has to be tested well I think. I cannot open the app since it cannot find chapter lists :/

File: app/build.gradle [hunk 2]
```diff
@@ -25,7 +25,7 @@ android {
 
     defaultConfig {
         minSdkVersion 10
-        targetSdkVersion 19
+        targetSdkVersion 21
```
Comment (stefanhoth) [inline app/build.gradle]: tumblr_m8tyfei8951qdfy6ao1_500

**Comment by stefanhoth** [comment]
@mauimauer Did you find the time to update the SSL cert on the hub so we can test the app again?

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -1,131 +1,193 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="org.gdg.frisbee.android" android:versionCode="27" android:versionName="1.0.10" android:installLocation="auto">
+  package="org.gdg.frisbee.android"
+  android:installLocation="auto">
```
Comment (stefanhoth) [inline app/src/main/AndroidManifest.xml]: `auto` can go as well as it is the default/ a non-settings.

**Comment by tasomaniac** [comment]
What do you guys think about proguard here? It does considerable amount of operation. We can disable it for Travis CI builds. Do you think if it affects the memory?

Whoever uploads to the play store dev console, they can enable proguard, build and upload.

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -1,131 +1,193 @@
...
+  <uses-permission android:name="android.permission.GET_ACCOUNTS" />
+  <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
+  <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
+  <uses-permission android:name="android.permission.WAKE_LOCK" />
+  <uses-permission android:name="android.permission.NFC" />
+  <uses-permission android:name="android.permission.GET_TASKS" />
 
-    <uses-feature
-        android:glEsVersion="0x00020000"
-        android:required="true"/>
```
Comment (stefanhoth) [inline app/src/main/AndroidManifest.xml]: Why was this required before? @mauimauer @friedger

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -1,131 +1,193 @@
...
-                android:resource="@xml/searchable"/>
-        </activity>
-        <activity android:name=".event.EventActivity"
-          android:parentActivityName=".activity.MainActivity">
-            <meta-data android:name="android.support.PARENT_ACTIVITY"
-                android:value=".activity.MainActivity"/>
-        </activity>
+  <uses-feature
+    android:glEsVersion="0x00020000"
+    android:required="true" />
```
Comment (stefanhoth) [inline app/src/main/AndroidManifest.xml]: Ah, it was just moved. but still, why is this required? @mauimauer @friedger

File: app/src/main/AndroidManifest.xml (hunk 4 above)
Comment (tasomaniac) [inline app/src/main/AndroidManifest.xml]: I think google maps requires it. I am not sure the app uses Google Maps V2 SDK though.

**Comment by stefanhoth** [comment]
@tasomaniac I think we should try it to find problems with proguard as well.

**Comment by HenriqueRocha** [comment]
I've rebased with @stefanhoth last commits as requested.
I left the battery charger at the office and it's already dead so I won't be able to do any more coding today.
The way I tested the app was to move the MAIN intent filter to the MainActivity and comment the check that checks if the "first start" flow was done.
I tested that all the activities had the new action bar.
I chose Material Blue as the action bar colour because blue is in the GDG icon but we can change that later as appropriate.

Should I do the changes that you're requesting as part of this PR or in my next PR to close #174 since those things in the Manifest were already there?

**Comment by tasomaniac** [comment]
Yey it passed. :)

File: app/src/main/AndroidManifest.xml (hunk 3 above)
Comment (friedger) [inline app/src/main/AndroidManifest.xml]: the documentation says other: internalOnly is the default http://developer.android.com/guide/topics/manifest/manifest-element.html#install

Comment (stefanhoth) [inline app/src/main/AndroidManifest.xml]: Ok, sorry, I misread. Taking it back then. 

On February 8, 2015 9:37:06 PM CET, "Friedger Müffke" <notifications@github.com> wrote:
>> @@ -1,131 +1,193 @@
>>  <?xml version="1.0" encoding="utf-8"?>
>>  <manifest xmlns:android="http://schemas.android.com/apk/res/android"
>> -    package="org.gdg.frisbee.android" android:versionCode="27"
>android:versionName="1.0.10" android:installLocation="auto">
>> +  package="org.gdg.frisbee.android"
>> +  android:installLocation="auto">
>
>the documentation says other: internalOnly is the default
>http://developer.android.com/guide/topics/manifest/manifest-element.html#install
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/gdg-x/frisbee/pull/180/files#r24302055

--
Sent from mobile. Please excuse brevity and any typos.

**Comment by stefanhoth** [comment]
This PR is still a bit too big for me to check right now. I will check it out and give it a spin. Will wait for the SSL update though. 

On February 8, 2015 9:23:25 PM CET, Said Tahsin Dane <notifications@github.com> wrote:
>Yey it passed. :)
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/gdg-x/frisbee/pull/180#issuecomment-73429556

--
Sent from mobile. Please excuse brevity and any typos.

File: app/src/main/java/org/gdg/frisbee/android/activity/SettingsActivity.java [hunk 6]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2013 The GDG Frisbee Project
+ * Copyright 2013-2015 The GDG Frisbee Project
```
Comment (friedger) [inline app/src/main/java/org/gdg/frisbee/android/activity/SettingsActivity.java]: We should remove the copyright per file to avoid these changes in the future. The license in the root folder is usually accepted as sufficient and valid for all files in the project.

File: app/src/main/res/values/gdg_themes.xml [hunk 7]
```diff
@@ -14,11 +14,16 @@
...
 
 <!-- Generated with http://android-holo-colors.com -->
 <resources xmlns:android="http://schemas.android.com/apk/res/android">
 
+  <style name="Theme.GDG.Base" parent="Theme.AppCompat.Light.DarkActionBar">
+    <item name="colorPrimary">@color/material_blue_500</item>
+    <item name="colorPrimaryDark">@color/material_blue_700</item>
+  </style>
+
   <style name="GDG.SeeTrough" parent="Theme.GDG">
```
Comment (friedger) [inline app/src/main/res/values/gdg_themes.xml]: could be moved below Theme.GDG to better understand the hierarchy.

File: app/src/main/java/org/gdg/frisbee/android/activity/SettingsActivity.java (hunk 6 above)
Comment (stefanhoth) [inline app/src/main/java/org/gdg/frisbee/android/activity/SettingsActivity.java]: :+1:

On February 8, 2015 9:44:21 PM CET, "Friedger Müffke" <notifications@github.com> wrote:
>> @@ -1,5 +1,5 @@
>>  /*
>> - * Copyright 2013 The GDG Frisbee Project
>> + * Copyright 2013-2015 The GDG Frisbee Project
>
>We should remove the copyright per file to avoid these changes in the
>future. The license in the root folder is usually accepted as
>sufficient and valid for all files in the project.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/gdg-x/frisbee/pull/180/files#r24302151

--
Sent from mobile. Please excuse brevity and any typos.

**Comment by friedger** [comment]
@HenriqueRocha Great effort! Thank you for your contributions.

If the PR contains formatting and code changes it might be good to highlight the important changes by leaving a comment in the code files. A screenshot for UI changes like this would be also a nice addition.
